mlflow/pyfunc/__init__.py

```diff
@@ -2507,8 +2507,22 @@ def batch_predict_fn(pdf, params=None):
                     return client.invoke(pdf).get_predictions()
 
             elif env_manager == _EnvManager.LOCAL:
+
+                def load_mlflow_model(model_uri, model_path, model_config):
+                    try:
+                        loaded_model = mlflow.pyfunc.load_model(
+                            model_path, model_config=model_config
+                        )
+                    except Exception:
+                        os.makedirs(model_path, exist_ok=True)
+                        loaded_model = mlflow.pyfunc.load_model(
+                            model_uri, dst_path=model_path, model_config=model_config
+                        )
+                    return loaded_model
+
                 if use_dbconnect_artifact:
                     model_path = dbconnect_artifact_cache.get_unpacked_artifact_dir(model_uri)
+                    loaded_model = load_mlflow_model(model_uri, model_path, model_config)
```

Comment:
Let's directly do
```
loaded_model = mlflow.pyfunc.load_model(model_path, model_config=model_config)
```
here,
because the model artifacts are already in `model_path` in this branch.